addons/mail/static/src/discuss/core/public_web/discuss_command_palette.js

```diff
@@ -0,0 +1,288 @@
+                this.onClickConfirm();
+                break;
+        }
+    }
+
+    async onClickConfirm() {
+        const name = this.state.name.trim();
+        if (!name) {
+            return;
+        }
+        await makeNewChannel(name, this.orm, this.store);
+        this.props.close();
+    }
+}
+
+class DiscussCommand extends Component {
+    static components = { ImStatus };
+    static template = "mail.DiscussCommand";
+    static props = {
+        counter: { type: Number, optional: true },
+        executeCommand: Function,
+        imgUrl: { String, optional: true },
+        name: String,
+        persona: { type: Object, optional: true },
+        channel: { type: Object, optional: true },
+        action: { type: Object, optional: true },
+        searchValue: String,
+        slots: Object,
+    };
+
+    setup() {
+        super.setup();
+        this.store = useState(useService("mail.store"));
+        this.ui = useState(useService("ui"));
+    }
+}
+
+// -----------------------------------------------------------------------------
+// add @ namespace + provider
+// -----------------------------------------------------------------------------
+commandSetupRegistry.add("@", {
+    debounceDelay: 200,
+    emptyMessage: _t("No conversation found"),
+    name: _t("conversations"),
+    placeholder: _t("Search a conversation"),
+});
+
+async function makeNewChannel(name, orm, store) {
+    const data = await orm.call("discuss.channel", "channel_create", [
+        name,
+        store.internalUserGroupId,
+    ]);
+    const { Thread } = store.insert(data);
+    const [channel] = Thread;
+    channel.open();
+}
+
+export class DiscussCommandPalette {
+    /**
+     * @param {import("@web/env").OdooEnv} env
+     * @param {import("services").ServiceFactories} env.services
+     */
+    constructor(env, options) {
+        this.env = env;
+        this.options = options;
+        this.dialog = env.services.dialog;
+        /** @type {import("models").Store} */
+        this.store = env.services["mail.store"];
+        this.orm = env.services.orm;
+        this.suggestion = env.services["mail.suggestion"];
+        this.ui = env.services.ui;
+        this.commands = [];
+        this.options = options;
+        this.cleanedTerm = cleanTerm(this.options.searchValue);
+    }
+
+    async fetch() {
+        await this.store.channels.fetch(); // FIXME: needed to search group chats without explicit name
+        await this.store.searchConversations(this.cleanedTerm);
+    }
+
+    /** @param {Record[]} [filtered] persona or thread to filters, e.g. being build already in a category in a patch such as MENTIONS or RECENT */
+    buildResults(filtered) {
+        const TOTAL_LIMIT = this.ui.isSmall ? 7 : 8;
+        const remaining = TOTAL_LIMIT - (filtered ? filtered.size : 0);
+        let personas = Object.values(this.store.Persona.records).filter(
+            (persona) =>
+                persona !== this.store.self &&
```

Comment:
self shouldn't be filtered out, it's a useful feature and users might have history there

Reply:
Ok to have it, but it's most likely best to make it very low priority, as this is a niche feature compared to search and making chat with other people